File: abstractions/php/src/Serialization/AdditionalDataHolder.php
<?php

namespace Microsoft\Kiota\Abstractions\Serialization;

interface AdditionalDataHolder {
    /**
     * Gets the additional data for this object that did not belong to the properties.
     * @return array<string,mixed> The additional data for this object.
     */
    public function getAdditionalData(): array;

    /**
     * Sets the additional data for this object that did not belong to the properties.
     * @param array<string, mixed> $value The additional data for this object.
     */
    public function setAdditionalData(array $value): void;
}

File: abstractions/java/lib/src/main/java/com/microsoft/kiota/serialization/AdditionalDataHolder.java
package com.microsoft.kiota.serialization;

import java.util.Map;

import javax.annotation.Nonnull;

/** Defines a contract for models that can hold additional data besides the described properties. */
public interface AdditionalDataHolder {
    /**
     * Gets the additional data for this object that did not belong to the properties.
     * @return The additional data for this object.
     */
    @Nonnull
    Map<String, Object> getAdditionalData();    
}
